### DISCUSSION TITLE
How to delete traces per each tenant?

### DISCUSSION THREAD

**Discussion by nima-ca** [grafana/tempo#4426] [Q&A]
Hi, is there any possibility to delete each trace for each individual tenant in tempo with HTTP API? I'm developing an application that let the user delete it's unused traces. If there are any ways to set retention policy or delete traces with HTTP API can be helpful. Or is there anyway to modify or delete data in backend storage is also helpful.

**Comment by mapno** [comment]
Hi @nima-ca. The `tempo-cli` has a command to delete traces by ID (see [docs](https://grafana.com/docs/tempo/latest/operations/tempo_cli/#drop-trace-by-id)), but there isn't an endpoint that specifically supports this.

**Reply by nima-ca** [reply]
Thanks. Is there any possibility to add HTTP API for such a thing? Or at least set retention policy for tenant with HTTP API in near future?

**Reply by mapno** [reply]
Nothing of the sort is planned in our roadmap, but we'll be glad to consider proposals.

**Reply by nima-ca** [reply]
thanks. I'll open up a feature request for that!
